lib/ex_aws/sqs/client.ex

```diff
@@ -0,0 +1,122 @@
+defmodule ExAws.SQS.Client do
+  use Behaviour
+
+  @moduledoc """
+  The purpose of this module is to surface the ExAws.SQS API tied to a single
+  configuration chose, such that it does not need passed in with every request
+
+  Usage:
+  ```elixir
+  defmodule MyApp.SQS do
+    use ExAws.SQS.Client, otp_app: :my_otp_app
+  end
+  ```
+
+  In your config
+  ```elixir
+  config :my_otp_app, ExAws,
+    sqs: [], #SQS config goes here
+  ```
+  """
+
+  @type sqs_permission :: 
+    :all | 
+    :receive_message | 
+    :delete_message |
+    :change_message_visibility | 
+    :get_queue_attributes 
+  @type sqs_acl :: [{binary, sqs_permission}]
```

Comment:
I want to avoid having types this because `[{"foo", 1}]` lacks the clarity of `%{"foo" => 1}` or `[foo: 1]` which have a clear key value aesthetic. If there isn't a good way to use atomic keys such that keyword list semantics apply, go with a map.